### PR TITLE
Show numeric keyboard on touch devices for DOB and salary inputs

### PR DESCRIPTION
**Pull request by niqjohnson**
Automatically pull up a numbers-only keyboard for DOB inputs and a numbers-and-symbols keyboard for the salary input on touch devices.

## Additions

- Add JavaScript to coerce touch devices to pull up the numbers-and-symbols keyboard for the salary input
- Add CSS to hide spinner controls for `<input type="number">`

## Changes

- Change DOB fields to `<input type="number">` with appropriate attributes

## Testing

- This definitely needs to be tested on **real** devices before getting merged (so maybe a Vagrant share):
  - Everything seems to be working well on my iPhone 6, but can someone test it out on other iPhones?
  - I don't have an Android phone to test on, so can someone check that out, too?

## Review

- @mistergone: This could use a good front-end code review

## Preview

Here's the numbers-only keyboard the DOB fields should pull up:

num

And here's the numbers-and-symbols keyboard for the salary field (since user testing showed having access to the comma was important):

num-sym

## Notes

- When the salary input is the first field tapped on on an iPhone, it's sometimes hidden by the keyboard (haven't figured out a good workaround).
- I had to do some JS magic to get browsers to remember a previously entered salary value if you tap on the field a second time, so you'll notice a brief flash of the text in that field in some cases.

### CODE REVIEW

**Comment by marteki** [comment]
Spooling up a virtualenv, pulling out my Nexus 5 (running Android Lollipop 5.1.1). Let the giant screencaps commence!

Let me try to enter my birth month:



Oooh! Now let me try to enter my birth day:



Birth year?



:clap: :clap: :clap: So excited!!! Salary?



Hmm. :confused: I clicked and nothing popped up.

Let me try again.



That's weird. :confused: A third attempt?



Aww. :cry:

**Comment by niqjohnson** [comment]
These fixes aren't working reliably, so I'm closing this PR. We'll figure out another way to help users on touch devices enter quick and accurate data.
